### PR TITLE
Termination of BlackInterpreter, update readme

### PR DESCRIPTION
**Pull request by RobinOppermann**
Fix the termination of the black interpreter

### CODE REVIEW

File: org.emoflon.ibex.gt.democles/src/org/emoflon/ibex/gt/democles/runtime/DemoclesGTEngine.java [hunk 1]
```diff
@@ -307,16 +312,40 @@ public void handleEvent(final MatchEvent event) {
...
+		if (matchList.isEmpty()) {
+			matches.remove(iDataFrame);
+		}
+	}
+	
+	/**
+	 * Removes the given match
+	 * @param match The match to remove
+	 */
+	@SuppressWarnings("unused")
```
Comment (patrickrobrecht) [inline org.emoflon.ibex.gt.democles/src/org/emoflon/ibex/gt/democles/runtime/DemoclesGTEngine.java]: Why does your PR contain unused code?

File: org.emoflon.ibex.gt.democles/src/org/emoflon/ibex/gt/democles/runtime/DemoclesGTEngine.java [hunk 2]
```diff
@@ -307,16 +312,40 @@ public void handleEvent(final MatchEvent event) {
...
 			}
 		});
 	}
+	
+	/**
+	 * Removes the match
+	 * @param iDataFrame The data frame the match is associated with
+	 * @param match The match to remove
+	 */
+	private void removeMatch(IDataFrame iDataFrame, IMatch match) {
```
Comment (patrickrobrecht) [inline org.emoflon.ibex.gt.democles/src/org/emoflon/ibex/gt/democles/runtime/DemoclesGTEngine.java]: This isn't correctly formatted. Crtl+Shift+F is your friend.

File: org.emoflon.ibex.gt.democles/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -23,3 +23,5 @@ Require-Bundle: org.eclipse.emf.ecore.xmi,
  org.gervarro.plan.dynprog;visibility:=reexport,
  org.gervarro.util;visibility:=reexport
 Automatic-Module-Name: org.emoflon.ibex.gt.democles
+Import-Package: gnu.trove.impl.hash,
+ gnu.trove.map.hash
```
Comment (patrickrobrecht) [inline org.emoflon.ibex.gt.democles/META-INF/MANIFEST.MF]: This import cannot be resolved when running the GT tests. It seems that the trove.jar is only imported in the project `org.emoflon.ibex.tgg.core.runtime`, which is not required at GT runtime.

File: README.md [hunk 4]
```diff
@@ -51,4 +52,6 @@ Running ```Testsuite_GLPK.launch``` requires GLPK (see installation step 5).
 	
 Running ```Testsuite_Gurobi.launch``` requires Gurobi (see installation step 6).
 
+Running ```Testsuite_CBC.launch``` requires Google OR tools (see installation step 7).
```
Comment (patrickrobrecht) [inline README.md]: References to installation steps are wrong.

File: org.emoflon.ibex.gt.democles/META-INF/MANIFEST.MF (hunk 3 above)
Comment (RobinOppermann) [reply org.emoflon.ibex.gt.democles/META-INF/MANIFEST.MF]: So which project should import the jar instead?

Comment (patrickrobrecht) [reply org.emoflon.ibex.gt.democles/META-INF/MANIFEST.MF]: I see two possiblities:
1. include another copy of the jar in `org.emoflon.ibex.gt.democles` (ugly of course)
2. move the jar to `org.emoflon.ibex.ibex.common` as this is the only common dependency of `org.emoflon.ibex.gt.democles` and `org.emoflon.ibex.tgg.core.runtime` if I haven't overseen anything. That looks a little ugly as well as `org.emoflon.ibex.ibex.common` imports a dependency it itself does not need.

Comment (anthonyanjorin) [reply org.emoflon.ibex.gt.democles/META-INF/MANIFEST.MF]: I would suggest (2).  The dependencies are fairly standard.  Trove will probably become necessary also for IBeX-GT as soon as we start optimising (any collection of matches should probably be replaced with something suitable from trove).

Comment (patrickrobrecht) [reply org.emoflon.ibex.gt.democles/META-INF/MANIFEST.MF]: That argument should valid.

Comment (RobinOppermann) [reply org.emoflon.ibex.gt.democles/META-INF/MANIFEST.MF]: Should be done with https://github.com/eMoflon/emoflon-ibex/pull/331

**Comment by patrickrobrecht** [comment]
https://github.com/eMoflon/emoflon-ibex/pull/331 fixed the problems at GT runtime.

**Pull request merged**